### PR TITLE
Fix CTZ fallback operator precedence

### LINKED ISSUE #2284: [BUG] Error in CTZ fallback implementation in `src/intrinsics.h`
**Describe the bug**

The `src/intrinsics.h` header implemented a fallback of CTZ (count-trailing-zero)  for compilers that aren't GCC or Clang:
https://github.com/valkey-io/valkey/blob/6d2dcdad4585edc40d0b58ac97e155fc511deb4b/src/intrinsics.h#L9-L18
But there is an operator precedence error in `while (value & 1 == 0)`, in C semantics this is interpreted as:
```
while (value & (1 == 0))
```
Which can be reduced to:
```
while (0)
```
So the loop is simply no-op and the fallback path will always return zero instead.

**To reproduce**

https://godbolt.org/z/Too1P4h81
Notice how clang, if `#else` path is taken, issues a warning about the precedence bug,
and how MSVC /O2 optimizer inlined the result as always zero (`xor edx, edx`)

**Expected behavior**

Correct calculation of CTZ.

**Additional information**

It seems that `src/intrinsics.h` is only used in `src/hyperloglog.c`, while there are other instances of direct `__builtin_*` usage in the codebase, unguarded by macro checks, at least these files:
```
deps/hdr_histogram/hdr_atomic.h
deps/hdr_histogram/hdr_histogram.c
src/dict.c
src/server.c
src/hashtable.c
```
So the codebase still requires a C compiler to implement or workaround these `__builtin_*` anyway, with or without this particular header, imo as a user, some consistency would be appreciated (that is, either maintain a complete and well-tested portability shim or go all-in with compiler builtins).

### PR DESCRIPTION
**Pull request by Taeknology**
## Summary

Fix the non-builtin `builtin_ctzll()` fallback so its loop tests the low bit correctly. Add a unit test that forces the fallback path and covers zero plus every single-bit position.

Fixes #2284

## Testing

- Forced-fallback unit test
- HyperLogLog integration tests
- Makefile and CMake TLS builds

### CODE REVIEW

File: src/unit/test_intrinsics.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+#include "generated_wrappers.hpp"
+
+#include <stdint.h>
+
+#if defined(__GNUC__) || defined(__clang__)
+#define INTRINSICS_TEST_HAS_DIAGNOSTIC_PRAGMA
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wparentheses"
```
[the remark this answers is not in the document]
Comment (Taeknology) [reply src/unit/test_intrinsics.cpp]: `src/unit/Makefile` adds `src/` through `-isystem`, so removing the local suppression alone does not expose warnings from `intrinsics.h`. Including it as `../intrinsics.h` keeps the fallback implementation covered by the existing `-Werror=parentheses` guard. A detached regression probe with the original expression fails compilation as expected.

File: src/unit/test_intrinsics.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#endif
+
+#undef __GNUC__
+#undef __clang__
```
[the remark this answers is not in the document]
Comment (Taeknology) [reply src/unit/test_intrinsics.cpp:16]: The pre-include `__INTRINSICS_H` check makes the fallback-path assumption explicit. Any earlier inclusion now fails compilation instead of silently leaving the builtin implementation active.

**Review by zuiderkwast — COMMENTED** [review]
The fallback code is wrong, but simply fixing the missing parentheses it was not what I suggested in the discussion in the issue.

The bigger picture is that we're using `__builtin_ctz`/`__builtin_clz` in many places throughout this code base without any fallbacks. The fallback is used in only one file: hyperloglog.c. That means that Valkey can't be compiled if these builtins are missing. Therefore, I prefer that we simply delete intrinsics.h and use the builtins instead.

To motivate keeping the fallbacks, we'd need to have fallbacks for all occurrences and have a CI job that compiles valkey with a compiler that doesn't have these builtins, so we'd catch any changes that doesn't use the fallbacks properly. (And if we do, we should use the C23 `<stdbit.h>` instead.) But that's a large change without a practical benefit.
